lighthouse-core/lib/network-request.js

```diff
@@ -321,6 +341,7 @@ class NetworkRequest {
     if (timing.requestTime === 0 || timing.receiveHeadersEnd === -1) return;
     // Take startTime and responseReceivedTime from timing data for better accuracy.
     // Timing's requestTime is a baseline in seconds, rest of the numbers there are ticks in millis.
+    // TODO: This skips the "queuing time" before the netstack has taken over ... is this a mistake?
     this.startTime = timing.requestTime;
```

Comment:
It seems this could be a mistake? startTime is initially set to the Network.requestWillBeSent timestamp AKA the left whisker .... then is reset here to be basically that + however long the "Queuing" time is (Pauls says that timings.requestTime is simply ignoring the time between renderer knowing about request and network manager knowing about request AKA queuing time)

Does this have implications on how we model our network records?

Reply:
really good catch.

In short: yeah i think this could be a problem.

After doing the [timing exercise](https://nettimings.surge.sh/Network-Timings.svg) id say our NetworkRequest class has too simple of a model.
Right now there's `startTime`, `responseReceivedTime`, `endTime`.  

My first inclination is to instead have `mainThreadStartTime`, `networkRequestTime`, `responseReceivedTime`, `networkEndTime`, and optionally `mainThreadEndTime`.

back to current state:

- Lantern slurps up the `startTime`, but i'm pretty sure the TCP stuff is simulated after that point, which is incorrect with this above assignment.
- uses-rel-preconnect is looking at network request starts vs timeOrigin, but it should be using the `mainThreadStartTime` aka `requestWillBeSent` timestamp instead.
- tags-blocking-first-paint calculates blocking time so it should def use mainThread- times rather than network-start times.

So yes... It's a mistake. And the fix looks kinda involved. :)  But also somewhat clear.